mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenProjectConfigurator.java

```diff
@@ -39,8 +44,25 @@ void configureTestRunner(Model model) {
 
             dependencyResolver.addRequiredDependencies(model);
 
+
+            final LocalStorage localStorage = new LocalStorage(Paths.get("").toFile());
             effectiveTestRunnerPluginConfigurations
-                .forEach(dependencyResolver::addAsPluginDependency);
+                .forEach(plugin -> {
+                    dependencyResolver.addAsPluginDependency(plugin);
```

Comment:
what if there are different versions for each plugin

Reply:
what do you mean about different versions of each plugin here, if I understand you correctly you mean what's happening if you have different versions of junit5 in the same project, which I think it is something really strange if this is what I think.

Reply:
Also notice that this plugin is added from a foreach which is all plugins `effectiveTestRunnerPlugin`

Reply:
I meant a different version of JUnit5 provider for surefire and failsafe...

Reply:
I think this is a really corner case, I mean I thought about this possibility but adding junit5 surefire platform with one version and in the same project another junit5 platform version for failsafe I would say that it is really really strange.

Reply:
There is another case - consider having surefire with JUnit 5 and failsafe without it...